backend/cmm_helpers.ml

```diff
@@ -3528,7 +3575,8 @@ let bigstring_load size unsafe mode arg1 arg2 dbg =
              bind "ba_data"
                (Cop (Cload (Word_int, Mutable), [field_address ba 1 dbg], dbg))
                (fun ba_data ->
-                 check_bound unsafe size dbg (bigstring_length ba dbg) idx
+                 check_bound unsafe size dbg (bigstring_data ba dbg)
```

Comment:
I think this needs to use "ba_data" (or "Cvar ba_data" or similar) not "bigstring_data ba dbg".  It's important to take extra care with this pattern, as re-using the expression rather than the bound variable could potentially cause a side effect to be duplicated (this has actually happened in production in the past).

Reply:
I see - it's fine to call `bigstring_length` directly because it's not duplicated, right? Or should I bind on that anyway?

Reply:
Actually I think I was wrong here, this was ok since `ba` was already bound using `bind` (but your new change looks better anyway).  I don't think we need to do anything about the `bigstring_length` calls.